src/ui/lp-render.ts

```diff
@@ -0,0 +1,236 @@
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*
+* */
+
+
+import {EditorView, ViewUpdate, Decoration, ViewPlugin, DecorationSet, WidgetType} from "@codemirror/view";
+import { EditorSelection, Range } from "@codemirror/state";
+import { syntaxTree } from "@codemirror/language";
+import {DataviewSettings} from "../settings";
+import { FullIndex } from "../data-index";
+import {/*Component,*/ editorLivePreviewField, TFile} from "obsidian";
+//import {asyncEvalInContext, DataviewInlineApi} from "../api/inline-api";
+import {DataviewApi} from "../api/plugin-api";
+import {renderMinimalDate, renderMinimalDuration, tryOrPropogate} from "../util/normalize";
+import {parseField} from "../expression/parse";
+import {executeInline} from "../query/engine";
+import {currentLocale} from "../util/locale";
+import {Literal, Values} from "../data-model/value";
+
+function selectionAndRangeOverlap(selection: EditorSelection, rangeFrom:
+    number, rangeTo: number) {
+
+    for (const range of selection.ranges) {
+        if ((range.from <= rangeTo) && (range.to) >= rangeFrom) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+
+// also returns text between inline code, so there always needs to be a check whether the correct prefix is used.
+function getInlineCodeBounds(view: EditorView, pos?: number): {start: number, end: number} | null {
+    const text = view.state.doc.toString()
+    if (typeof pos === "undefined") {
+        pos = view.state.selection.main.from;
+    }
+    let left = text.lastIndexOf('`', pos)
+    const right = text.indexOf('`', pos)
+    // no backtick before or after the current backtick
+    if (left === -1 || right === -1) return null;
+    const leftNewline = text.lastIndexOf('\n', pos)
+    const rightNewline = text.indexOf('\n', pos)
+
+    // start or end of document w/o new lines
+    if (leftNewline === -1 || rightNewline === -1) {
+        return {start: left , end: right+1}
+    }
+
+    if (leftNewline > left || rightNewline < right) return null;
+
+    return {start: left , end: right+1}
+}
+
+
+
+
+class InlineWidget extends WidgetType {
+    constructor(readonly markdown: string | Literal, readonly settings: DataviewSettings, readonly currentFile: TFile) {
+        super();
+    }
+    eq(other: InlineWidget): boolean {
+        return other.markdown === this.markdown;
+    }
```

Comment:
Creating an empty element and appending to it for handling async query rendering or JS would probably stop this from working.

Which should result in more redrawing of these components.
https://codemirror.net/docs/ref/#view.WidgetType.eq